### PR TITLE
Automate CR configuration documentation

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Automate CR configuration documentation

## How can this be tested?

`make doc/api-ref`

Compare with official documentation https://docs.dynatrace.com/docs/setup-and-configuration/setup-on-k8s/reference/dynakube-parameters

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: doc/dynakube-api-ref.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+DynaKube is the Schema for the DynaKube API
+
+## DynaKube is the Schema for the DynaKube API
```
Comment (0sewa0) [inline doc/dynakube-api-ref.md:3]: duplicate + does not say much 

`DynaKube Schema` would work fine

File: doc/edgeconnect-api-ref.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+EdgeConnect is the Schema for the EdgeConnect API
+
+## EdgeConnect is the Schema for the EdgeConnect API
```
Comment (0sewa0) [inline doc/edgeconnect-api-ref.md:3]: duplicate + does not say much

`EdgeConnect Schema` would be enough

File: hack/make/doc/api-ref.mk [hunk 3]
```diff
@@ -0,0 +1,3 @@
+doc/api-ref: manifests
+	python3 ./hack/doc/custom_resource_params_to_md.py ./config/crd/bases/dynatrace.com_dynakubes.yaml > ./doc/dynakube-api-ref.md
+	python3 ./hack/doc/custom_resource_params_to_md.py ./config/crd/bases/dynatrace.com_edgeconnects.yaml > ./doc/edgeconnect-api-ref.md
```
Comment (0sewa0) [inline hack/make/doc/api-ref.mk:3]: would it make sense to start to have folders in the `doc` folder 🤔 

these could go into `/doc/api/...` for example

Comment (andriisoldatenko) [reply hack/make/doc/api-ref.mk:3]: yeah good idea, will do it.

File: doc/edgeconnect-api-ref.md (hunk 2 above)
Comment (andriisoldatenko) [reply doc/edgeconnect-api-ref.md:3]: agree on it.

Comment (andriisoldatenko) [reply doc/edgeconnect-api-ref.md]: fixed in 8eca23cd

File: doc/dynakube-api-ref.md (hunk 1 above)
Comment (andriisoldatenko) [reply doc/dynakube-api-ref.md]: fixed in 8eca23cd

**Review by 0sewa0 — APPROVED** [review]
practically LGTM, ran it locally

minor nitpick: do we have it somewhere that we need to `pip3 install pyyaml`, so that people not that versed in python can also run it no problem?

**Comment by andriisoldatenko** [comment]
> minor nitpick: do we have it somewhere that we need to `pip3 install pyyaml`, so that people not that versed in python can also run it no problem?

Do you want me update https://github.com/dynatrace/dynatrace-operator/blob/79c807834432dc170c89d9a87f274ffce1cccbe6/hack/doc/README.md#usage ?

**Review by 0sewa0 — APPROVED** [review]

**Comment by albertogdd** [comment]
About python scripts, I'm only scared about siloing them, because I believe not everyone in the team has python knowledge (I don't). Just something to think about.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -113,6 +113,24 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version-file: "${{ github.workspace }}/go.mod"
+      - name: Run make api docs to see uncommitted changes
+        run: make doc/api-ref
+      - name: Check for uncommitted changes
+        id: check-changes
+        uses: mskri/check-uncommitted-changes-action@v1.0.1
```
Comment (chrismuellner) [inline .github/workflows/ci.yaml:129]: pinned version?

File: hack/doc/README.md [hunk 5]
```diff
@@ -31,3 +33,25 @@ make doc/permissions
 python3 <operator-repo>/hack/doc/role-permissions2md.py https://github.com/Dynatrace/dynatrace-operator/releases/download/v0.12.0/openshift.yaml
 python3 <operator-repo>/hack/doc/role-permissions2md.py https://github.com/Dynatrace/dynatrace-operator/releases/download/v0.12.0/openshift-csi.yaml
 ```
+
+## custom_resource_params_to_md.py
+
+This script generates API docs for custom resources
+
+Pre-requisites:
```
Comment (chrismuellner) [inline hack/doc/README.md:41]: prerequisites are formatted differently in the previous script. Let's try to align them

**Review by chrismuellner — CHANGES_REQUESTED** [review]

**Pull request merged**